### ISSUE TITLE
[Discussion] How can increase outside contributions and reduce entry to the code?

### ISSUE THREAD

**Issue by fokusferit** [webpack/webpack-cli#180]
This Issue is about the following:

- How can we increase first time contributions ? 
  - What kind of onboarding strategy can we provide people so it is easier to catch up with the code, the architecture ?
  - I think we need to have a broader strategy here with Action Items, e.g. Cleaner Code blocks, Documentation of some core concepts or Some basic Hand-Ons text like "Where to start if you want to implement feature X or fix bug Y". 
- How can we increase our process of merging PR's faster ?
  - Is also related to attracting more contributors, but I think this is also about how can we be faster in reviewing PR's and giving feedback faster. Basically lowering the time of first contact.

**Comment by thescientist13** [comment]
I have a couple modest thoughts.  😊 

###  webpack Fundamentals
I think directing contributors to a good overview of the webpack codebase in general would be helpful, something to orient themselves on the general patterns, conventions, and practices of the underlying tool, e.g. webpack itself.  Something along the lines of the webpack [Concepts](https://webpack.js.org/concepts/) documentation. and something similar for the CLI as well.

### Roadmap
Something that communicates the motivations of the project, like what is being proposed in in #181 .  I know there is already an issue for it, but I know for me, knowing what the short and long term goals (e.g. developer use cases) of the CLI helps.  

I think knowing the stance of the CLI on convention over configuration becomes an important detail to communicate.

 ### CLI case study / developer guides
Maybe do a case study on a core part of the code and do a case study or document how a feature was developed / delivered?  Something that can show new developers the thought process behind the architecture and technical design?

**Comment by fokusferit** [comment]
Thanks for the feedback / ideas @thescientist13 .
I agree with you, I think one of the first steps would be really to update the Readme and add things like:

- Purpose / Motivation behind it
- Getting started (points to some initial reads around webpack)
- Roadmap (initially this should reflect some of GH issues, so redirecting to issues page with a label "next steps" or "next release".
- How to use / Case studies. 

I will open a [WIP] PR tomorrow and hopefully we can all start working on that.

**Comment by ematipico** [comment]
To add it the list, I think something nice is to understand how to create/modify tests.
Tests are useful because they help developers to understand how a single functionality works (or part of it).

**Comment by ev1stensberg** [comment]
@ematipico also, we need to add those before we can merge into webpack/core
